trin-storage/src/versioned/utils.rs

```diff
@@ -0,0 +1,183 @@
+use ethportal_api::types::portal_wire::ProtocolId;
+use r2d2::PooledConnection;
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{named_params, OptionalExtension};
+use trin_metrics::{portalnet::PORTALNET_METRICS, storage::StorageMetricsReporter};
+
+use crate::{error::ContentStoreError, PortalStorageConfig};
+
+use super::{sql, store::VersionedContentStore, ContentType, StoreVersion};
+
+/// Ensures that the correct version of the content store is used (by migrating the content if
+/// that's not the case).
+pub fn create_store<S: VersionedContentStore>(
+    content_type: ContentType,
+    network: ProtocolId,
+    config: PortalStorageConfig,
+) -> Result<S, ContentStoreError> {
+    let conn = config.sql_connection_pool.get()?;
+
+    let old_version = lookup_store_version(content_type, &conn)?;
```

Comment:
We should probably have some logic to detect legacy version (e.g. `Legacy*`).

It's not important until we have implementations.

Once this PR is merged, I will create an Issue to track this.